lib/carin_for_blue_button_test_kit/carin_search_test.rb

```diff
@@ -28,53 +28,249 @@ def all_scratch_resources
-        returned_resources = []
-        param_value.each do |id|
-          search_params[param_name] = id
-          returned_resources.concat(perform_search(search_params))
+      all_scratch_resources.concat(returned_resources).uniq! if first_search?
+      save_delayed_references(returned_resources) if saves_delayed_references?
+      perform_response_validation(returned_resources, search_params)
+    end
+
+    def execute_search(search_params)
+      if patient_id_param?(search_param_names[0])
+        patient_id_list.inject([]) do |resources, id|
+          search = { search_param_names[0] => id }
+          perform_search_and_validate(search)
+          resources.concat(extract_relevant_resources)
         end
-        search_params[param_name] = param_value.join(',')
       else
-        search_params[param_name] = param_value
-        returned_resources = perform_search(search_params)
+        perform_search_and_validate(search_params)
+        extract_relevant_resources
       end
-
-      all_scratch_resources.concat(returned_resources).uniq! if first_search?
-
-      perform_response_validation(returned_resources, search_params)
     end
 
-    def perform_search(search_params)
+    def perform_search_and_validate(search_params)
       fhir_search(resource_type, params: search_params)
 
       assert_response_status(200)
       assert_resource_type(:bundle)
+    end
 
+    def extract_relevant_resources
       extract_resources_from_bundle(bundle: resource, response:).select do |item|
         item.resourceType == resource_type
       end
     end
 
+    def build_search_params(param_value = nil)
+      search_params = {}
+      param_name = search_param_names[0]
+
+      if patient_id_param?(param_name)
+        search_params[param_name] = patient_id_list.join(',')
+      else
+        param_value = find_param_value(param_value, param_name) if param_value.blank?
+        skip_if param_value.blank?, no_resources_skip_message
+        search_params[param_name] = param_value
+      end
+
+      search_params
+    end
+
+    def find_param_value(param_value, param_name)
+      all_scratch_resources.each do |resource|
+        break if param_value.present?
+
+        param_value = search_param_value(param_name, resource)
+      end
+
+      if param_value.nil? && param_name == '_id' && resource_type != 'ExplanationOfBenefit'
+        resources = readable_resources(scratch.dig(:references, resource_type))
+        param_value = resource_id(resources.first)
+      end
+
+      param_value
+    end
+
+    def search_param_value(name, resource, include_system = false)
+      paths = determine_paths(name)
+      paths.each do |path|
+        element = find_a_value_at(resource, path) { |el| element_has_valid_value?(el, include_system) }
+        search_value = process_element(name, element, include_system, resource)
+        return search_value.gsub(',', '\\,') if search_value.present?
+      end
+      nil
+    end
```

Comment:
When running the _lastUpdated test, it seems like the formatting for date is getting messed up somewhere. When I ran the _lastUpdated test for patient when using the cpcds server (with Patient1 as the patient id), the lastUpdated value in meta.lastUpdated is `2022-09-17T20:32:19.462+00:00`, and when the lastUpdated test is ran, `2022-09-17T20:32:19.462%2B00:00` is passed in as the date, which gets an error due to the param not being of type date

Reply:
The problem was with the server. It has an interceptor that checked only if the input was in date format, thus rejecting date time format. I was the one who implemented that checks on the server side. I fixed it to accept date time/offset date time format ass well.